### PR TITLE
Reset node selector of managed namespaces on OCP3

### PR DESCRIPTION
**Pull request by thbkrkr**
When the environment is OCP3, execute `oc annotate --overwrite namespace <ns> openshift.io/node-selector=""` for all managed namespaces to prevent that daemonset pods from being scheduled on some nodes, because by default there can be a node selector at the OCP project level.

Relates to #4128.

### CODE REVIEW

File: test/e2e/cmd/run/run.go [hunk 1]
```diff
@@ -377,14 +377,28 @@ func (h *helper) createOperatorNamespaces() error {
 }
 
 func (h *helper) createManagedNamespaces() error {
-	log.Info("Creating managed namespaces")
+	log.Info("Creating managed namespacesS")
```
Comment (barkbay) [inline test/e2e/cmd/run/run.go:380]:
```suggestion
	log.Info("Creating managed namespaces")
```

File: test/e2e/cmd/run/run.go [hunk 2]
```diff
@@ -377,14 +377,28 @@ func (h *helper) createOperatorNamespaces() error {
...
-		_, err := h.kubectlApplyTemplate("config/e2e/managed_namespaces.yaml", h.testContext)
-		return err
+		_, err = h.kubectlApplyTemplate("config/e2e/managed_namespaces.yaml", h.testContext)
+	} else {
+		err = h.kubectlApplyTemplateWithCleanup("config/e2e/managed_namespaces.yaml", h.testContext)
 	}
 
-	return h.kubectlApplyTemplateWithCleanup("config/e2e/managed_namespaces.yaml", h.testContext)
+	// Reset the node selector for all managed namespaces to override any possible OCP project node selector that might
+	// prevent scheduling daemonset pods on some nodes
```
Comment (barkbay) [inline test/e2e/cmd/run/run.go:390]:
```suggestion
	// prevent scheduling daemonset pods on some nodes.
```

File: test/e2e/cmd/run/run.go [hunk 3]
```diff
@@ -377,14 +377,28 @@ func (h *helper) createOperatorNamespaces() error {
...
+	if h.testContext.Ocp3Cluster {
+		log.Info("Resetting namespace node selector")
+		for _, ns := range h.testContext.Operator.ManagedNamespaces {
+			if err := exec.Command("oc", "annotate", "--overwrite", "namespace", ns, "openshift.io/node-selector=").Run(); err != nil {
+				log.Info("err:", err)
```
Comment (barkbay) [inline test/e2e/cmd/run/run.go:395]:
```suggestion
				log.Error(err, "Error while setting openshift.io/node-selector annotation", "namespace", ns)
```

**Review by barkbay — APPROVED** [review]
lgtm

File: test/e2e/cmd/run/run.go (hunk 3 above)
Comment (thbkrkr) [reply test/e2e/cmd/run/run.go]: Sorry I forgot to remove this _debug_ logger.
Fixed with 61911e2.

**Pull request merged**